examples/storm-starter/src/jvm/org/apache/storm/starter/trident/TridentHBaseWindowingStoreTopology.java

```diff
@@ -0,0 +1,122 @@
+import org.apache.storm.trident.testing.CountAsAggregator;
+import org.apache.storm.trident.testing.FixedBatchSpout;
+import org.apache.storm.trident.tuple.TridentTuple;
+import org.apache.storm.trident.windowing.WindowsStoreFactory;
+import org.apache.storm.trident.windowing.config.TumblingCountWindow;
+import org.apache.storm.tuple.Fields;
+import org.apache.storm.tuple.Values;
+import org.apache.storm.utils.Utils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ *
+ */
+public class TridentHBaseWindowingStoreTopology {
+    private static final Logger LOG = LoggerFactory.getLogger(TridentHBaseWindowingStoreTopology.class);
+
+    public static StormTopology buildTopology(WindowsStoreFactory windowsStore) throws Exception {
+        FixedBatchSpout spout = new FixedBatchSpout(new Fields("sentence"), 3, new Values("the cow jumped over the moon"),
+                new Values("the man went to the store and bought some candy"), new Values("four score and seven years ago"),
+                new Values("how many apples can you eat"), new Values("to be or not to be the person"));
+        spout.setCycle(true);
+
+        TridentTopology topology = new TridentTopology();
+
+        Stream stream = topology.newStream("spout1", spout).parallelismHint(16).each(new Fields("sentence"),
+                new Split(), new Fields("word"))
+                .window(TumblingCountWindow.of(1000), windowsStore, new Fields("word"), new CountAsAggregator(), new Fields("count"))
+//                .tumblingTimeWindow(new BaseWindowedBolt.Duration(3, TimeUnit.SECONDS), windowsStore, new Fields("word"), new CountAsAggregator(), new Fields("count"))
+                .each(new Fields("count"), new Debug())
+                .each(new Fields("count"), new Echo(), new Fields("ct"));
+
+        return topology.build();
+    }
+
+    public static class Split extends BaseFunction {
+        @Override
+        public void execute(TridentTuple tuple, TridentCollector collector) {
+            String sentence = tuple.getString(0);
+            for (String word : sentence.split(" ")) {
+                collector.emit(new Values(word));
+            }
+        }
+    }
+
+    static class Echo implements Function {
+
+        @Override
+        public void execute(TridentTuple tuple, TridentCollector collector) {
+            LOG.info("##########Echo.execute: " + tuple);
+            collector.emit(tuple.getValues());
+        }
+
+        @Override
+        public void prepare(Map conf, TridentOperationContext context) {
+
+        }
+
+        @Override
+        public void cleanup() {
+
+        }
+    }
+
+    public static void main(String[] args) throws Exception {
+        Config conf = new Config();
+        conf.setMaxSpoutPending(20);
+        conf.put(Config.TOPOLOGY_TRIDENT_WINDOWING_INMEMORY_CACHE_LIMIT, 2);
+
+        // window-state table should already be created with cf:tuples column
```

Comment:
Is it a pre-req for using HBaseWindowsStoreFactory that the window-state table should be created with cf:tuples ? Is it documented somewhere ? Its better if this can be auto-created.